lib/sycamore/sycamore/transforms/detr_partitioner.py

```diff
@@ -469,29 +480,39 @@ def _convert_bbox_coordinates(
         y2 = height - y2
         return x1, y1, x2, y2
 
-    def extract(self, filename: Union[str, IOBase]) -> List[List[Element]]:
+    def extract(self, filename: Union[str, IOBase], use_cache=True) -> List[List[Element]]:
         # The naming is slightly confusing, but `open_filename` accepts either
         # a filename (str) or a file-like object (IOBase)
-        with open_filename(filename, "rb") as fp:
-            fp = cast(BinaryIO, fp)
-            pages = []
-            for page, page_layout in self._open_pdfminer_pages_generator(fp):
-                width = page_layout.width
-                height = page_layout.height
-                texts: List[Element] = []
-                for obj in page_layout:
-                    x1, y1, x2, y2 = self._convert_bbox_coordinates(obj.bbox, height)
-
-                    if hasattr(obj, "get_text"):
-                        text = Element()
-                        text.type = "text"
-                        text.bbox = BoundingBox(x1 / width, y1 / height, x2 / width, y2 / height)
-                        text.text_representation = obj.get_text()
-                        if text.text_representation:
-                            texts.append(text)
-
-                pages.append(texts)
-            return pages
+
+        hash_key = pdf_miner_cm.get_hash_key(filename)
```

Comment:
Let's add a TODO to see if we can share the file content used here to compute the hash with the PDF miner routine happening below in the event of a cache miss.